portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,52 +26,62 @@
 import com.google.gson.Gson;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Date;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that contains saves and gets comments from database. */
+/** Servlet that contains, saves and gets comments from database. 
+ *  Uses LoginServlet to determine if a user is logged in to
+ *  either show or hide comments.
+ */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private List<String> comments = new ArrayList<>();
     private final Gson gson = new Gson();
+    private String currentUserEmail;
```

Comment:
Data stored in the Servlet is shared across different calls to the servlet, including from different users.

If you deploy this implementation, try this: share the page link with your podmates, then have everyone log in. Once everyone is logged in, have everyone post a comment. It is very likely that you will see all three comments appear as being posted by only one user, because all requests to the server will use the most-recently set user from doGet.

In general, you should not store request-specific information on the servlet like this, as there might be another request processing at the same time.